library/src/main/java/org/apache/apex/malhar/lib/window/impl/AbstractWindowedOperator.java

```diff
@@ -389,8 +397,32 @@ public void processWatermark(ControlTuple.Watermark watermark)
   @Override
   public void setup(Context.OperatorContext context)
   {
-    this.windowWidthMillis = context.getValue(Context.DAGContext.STREAMING_WINDOW_SIZE_MILLIS);
+    this.timeIncrement = context.getValue(Context.OperatorContext.APPLICATION_WINDOW_COUNT) * context.getValue(Context.DAGContext.STREAMING_WINDOW_SIZE_MILLIS);
     validate();
+    windowStateMap.setup(context);
```

Comment:
I'm thinking if we could provide an automatic way to call these method by set annotation on the field just like those IoC frameworks do